File: C++/Simple 2D extensible engine/Shooter Game/Enemy.cpp
#include "Enemy.h"

Enemy::Enemy ( Player* newPlayer, RenderShape* shape ) :GameObject ( shape ) { player = newPlayer; }

void Enemy::Update ( float deltaTime )
{
	Vector3 diff = ( player->GetTransform ().Position - transform.Position ).Normalize ();
	transform.Position += diff * deltaTime * speed;
}

void Enemy::TakeDamage ( int damage ) { health -= damage; }

int Enemy::GetHealth () { return health; }

File: Unity Games/2020 3D School Assignments/3D Physics Based Game/Enemy.cs
/*
* Greg Brandt
* Assignment 7
* Controls enemy behavior
*/
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Enemy : MonoBehaviour
{
    [SerializeField] float speed = 3.0f;
    Rigidbody enenmyRb;
    GameObject player;

    // Start is called before the first frame update
    void Start()
    {
        enenmyRb = GetComponent<Rigidbody>();
        player = GameObject.Find("Player");
    }

    // Update is called once per frame
    void Update()
    {
        Vector3 lookDirection = (player.transform.position - transform.position).normalized;
        enenmyRb.AddForce(lookDirection * speed);
        if(transform.position.y < -10)
        {
            Destroy(gameObject);
        }
    }
}
